src/browser/api/webcontents.ts

```diff
@@ -0,0 +1,90 @@
+import * as url from 'url';
+
+
+export function executeJavascript(webContents: Electron.WebContents, code: string, callback: (e: any, result: any) => void): void {
+    webContents.executeJavaScript(code, true, (result) => {
+        callback(undefined, result);
+    });
+}
+
+export function getInfo(webContents: Electron.WebContents) {
+    return {
+        canNavigateBack: webContents.canGoBack(),
+        canNavigateForward: webContents.canGoForward(),
+        title: webContents.getTitle(),
+        url: webContents.getURL()
+    };
+}
+
+export function getAbsolutePath(webContents: Electron.WebContents, path: string) {
+    const windowURL = webContents.getURL();
+
+    return  url.resolve(windowURL, path);
+}
+
+export function navigate (webContents: Electron.WebContents, url: string) {
+    // todo: replace everything here with "return webContents.loadURL(url)" once we get to electron 5.*
+    // reason: starting electron v5, loadUrl returns a promise that resolves according to the same logic we apply here
+    const navigationEnd = createNavigationEndPromise(webContents);
+    webContents.loadURL(url);
+    return navigationEnd;
+}
+
+export async function navigateBack (webContents: Electron.WebContents) {
+    if (!webContents.canGoBack()) {
+        throw new Error('Cannot navigate back');
+    }
+    const navigationEnd = createNavigationEndPromise(webContents);
+    webContents.goBack();
+    return navigationEnd;
+}
+
+export async function navigateForward (webContents: Electron.WebContents) {
+    if (!webContents.canGoForward()) {
+        const error = new Error('Cannot navigate forward');
+        throw error;
+    }
+    const navigationEnd = createNavigationEndPromise(webContents);
+    webContents.goForward();
+    return navigationEnd;
+}
+
+export function getZoomLevel(webContents: Electron.WebContents, callback: (zoomLevel: number) => void) {
+    webContents.getZoomLevel(callback);
```

Comment:
it would've been nice if this used a promise instead of a cb

Reply:
That's not how the electron api works for this one

Reply:
I'm suggesting returning a promise that resolves when the call is done, like so:

> return new Promise(res => webContents.getZoomLevel(res));

Reply:
I agree it's better, but I'm trying to stick as close to the window.js code as I can. It's already a pretty large change set and I want to keep it as safe as I can.

Reply:
makes sense.